lib/plugin-loader.js

```diff
@@ -858,107 +858,111 @@ PluginLoader.prototype = {
           count: newPlugins.length
         });
       }
-  },
+  }),
 
   // Note - Not to be confused with auth capabilities, that describe what an auth plugin can do
-  async getComponentCapabilities(config) {
-    let appServerComp = {};
-    let agentComp = {};
-    let host;
-    let port;
+  getComponentCapabilities(config) {
+    return new Promise((complete, fail)=> {
+      let appServerComp = {};
+      let agentComp = {};
+      let host;
+      let port;
 
-    appServerComp.os = process.platform; // Operating system
-    appServerComp.cpu = process.arch; // CPU architecture
+      appServerComp.os = process.platform; // Operating system
+      appServerComp.cpu = process.arch; // CPU architecture
 
-    if (config && config.agent.host) {
-      host = config.agent.host;
-      if (config.agent.http) {
-        port = config.agent.http.port;
-      } else if (config.agent.https) {
-        port = config.agent.https.port;
+      if (config && config.agent.host) {
```

Comment:
The `agent` section is optional in `server.json`. I would check `config && config.agent && config.agent.host`.